R/plotGMA.R

```diff
@@ -0,0 +1,270 @@
+                 palette = c("chartreuse4",  "firebrick3"))
+        save_pdf(met_heatmap, 
+                 height = 15, 
+                 width = 35, 
+                 units = "cm", 
+                 filename = paste(additionalFilename,
+                                  "heatmap_genelist_met.pdf", 
+                                  sep = ""))
+      
+      # If genelist only contain one oncogenic mediator gene type
+    } else {
+      
+      # Create heatmap of genes in genelist
+      met_heatmap <- summary_heatmap %>% 
+        heatmap(.data = .,
+                .row = Met_type,
+                .column = Hugo_Symbol, 
+                .value = Met_count,
+                cluster_rows = FALSE,
+                cluster_columns = TRUE,
+                show_column_dend = FALSE,
+                show_column_names = TRUE,
+                scale = "none",
+                palette_value = c("white", "blue", "darkblue"),
+                column_title = "Heatmap of methylation state by EpiMix")  %>% 
+        add_tile(logFC, 
+                 palette = c("chartreuse4",  "firebrick3")) 
+        save_pdf(met_heatmap, 
+                 height = 15, 
+                 width = 35, 
+                 units = "cm", 
+                 filename = paste(additionalFilename,
+                                  "heatmap_genelist_met.pdf", 
+                                  sep = ""))
+      
+    }
+    
+  }
+  
+  ## Complete mode
+  else if (type == "complete") {
+    
+    # Heatmap of all genes grouped by oncogenic mediator types
+    met_heatmap <- summary_heatmap %>% 
+      group_by(Moonlight_Oncogenic_Mediator) %>% 
+      heatmap(.data = .,
+              .row = Met_type,
+              .column = Hugo_Symbol, 
+              .value = Met_count,
+              cluster_rows = FALSE,
+              cluster_columns = TRUE,
+              show_column_dend = FALSE,
+              show_column_names = TRUE,
+              scale = "none",
+              palette_value = c("white", "blue", "darkblue"),
+              column_title = "Heatmap of methylation state by EpiMix") %>% 
+      add_tile(Moonlight_Oncogenic_Mediator, 
+               palette = c("goldenrod2", "dodgerblue3")) %>% 
+      add_tile(logFC, 
+               palette = c("chartreuse4",  "firebrick3")) 
+      save_pdf(met_heatmap, 
+               height = 15, 
+               width = 35, 
+               units = "cm", 
+               filename = paste(additionalFilename,
+                                "heatmap_complete_met.pdf", 
+                                sep = ""))
+    
+  }
+  
+  ## Split mode
+  else if(type == "split") {
+    
+    if (dir.exists("./heatmaps")) {
+      stop("'heatmaps' folder already exits. Please remove it.")
+    }
+    else {
+      print("A temporary heatmaps folder is created. It will be removed before finalising.")
+      dir.create(path = "./heatmaps", 
+                 showWarnings = TRUE, 
+                 recursive = TRUE)
+    }
```

Comment:
if you need a temporary folder it would be [better practice](https://www.jdtrat.com/blog/temporary-storage-in-r/) to use `tempdir()` to get one, so you don't have to manage it and it's safer

Reply:
updated